### PR TITLE
Bring the latest extension::DrawNodeEx to axmol

### PR DESCRIPTION
**Pull request by aismann**
## Describe your changes

DrawNodeEx 0.91

### For each PR
- [ ] Add Copyright if it missed:   
      - `"Copyright (c) 2019-present Axmol Engine contributors (see AUTHORS.md)."`
- [x] I have performed a self-review of my code.
       
   Optional:
   - [ ] I have checked readme and add important infos to this PR.
   - [ ] I have added/adapted some tests too.
          
### For core/new feature PR
- [ ] I have checked readme and add important infos to this PR.
- [ ] I have added thorough tests.

### CODE REVIEW

**Comment by aismann** [comment]
@halx99 
build / osx-x64 (pull_request) Failing after 12m
Why?
Seems 1 unit test failed (but its not DrawNodeEx, right?)

File: extensions/DrawNodeEx/src/DrawNodeEx/DrawNodeEx.cpp [hunk 1]
```diff
@@ -166,7 +174,7 @@ DrawNodeEx* DrawNodeEx::create(float defaultLineWidth)
     return ret;
 }
 
-void DrawNodeEx::ensureCapacity(int count)
+void DrawNodeEx::ensureCapacityGLTriangle(int count)
```
Comment (halx99) [inline extensions/DrawNodeEx/src/DrawNodeEx/DrawNodeEx.cpp]:
```suggestion
void DrawNodeEx::ensureCapacityTriangle(int count)
```

File: extensions/DrawNodeEx/src/DrawNodeEx/DrawNodeEx.cpp [hunk 2]
```diff
@@ -215,39 +223,39 @@ bool DrawNodeEx::init()
 {
     _blendFunc = BlendFunc::ALPHA_PREMULTIPLIED;
     updateShader();
-    ensureCapacity(512);
+    ensureCapacityGLTriangle(512);
```
Comment (halx99) [inline extensions/DrawNodeEx/src/DrawNodeEx/DrawNodeEx.cpp]:
```suggestion
    ensureCapacityTriangle(512);
```

File: extensions/DrawNodeEx/src/DrawNodeEx/DrawNodeEx.cpp [hunk 3]
```diff
@@ -657,39 +666,37 @@ void DrawNodeEx::drawCatmullRom(ax::PointArray* points, unsigned int segments, c
 void DrawNodeEx::drawDot(const Vec2& pos, float radius, const Color4B& color)
 {
     unsigned int vertex_count = 2 * 3;
-    ensureCapacity(vertex_count);
+    ensureCapacityGLTriangle(vertex_count);
```
Comment (halx99) [inline extensions/DrawNodeEx/src/DrawNodeEx/DrawNodeEx.cpp]:
```suggestion
    ensureCapacityTriangle(vertex_count);
```

File: extensions/DrawNodeEx/src/DrawNodeEx/DrawNodeEx.cpp [hunk 4]
```diff
@@ -725,92 +727,143 @@ void DrawNodeEx::drawSegment(const Vec2& from, const Vec2& to, float radius, con
     Vec2 v6 = a - (nw - tw);
     Vec2 v7 = a + (nw + tw);
 
+    int ii = 0;
+    unsigned int vertex_count = 6 * 3;
+
+    ensureCapacityGLTriangle(vertex_count);
```
Comment (halx99) [inline extensions/DrawNodeEx/src/DrawNodeEx/DrawNodeEx.cpp]:
```suggestion
    ensureCapacityTriangle(vertex_count);
```

File: extensions/DrawNodeEx/src/DrawNodeEx/DrawNodeEx.cpp [hunk 5]
```diff
@@ -1040,11 +1092,11 @@ void DrawNodeEx::_drawPolygon(const Vec2* verts,
     Color4B borderColor = borderColo;
 
     auto triangle_count = outline ? (3 * count - 2) : (count - 2);
-    auto vertex_count   = 3 * triangle_count;
-    ensureCapacity(vertex_count);
+    auto vertex_count = 3 * triangle_count;
+    ensureCapacityGLTriangle(vertex_count);
```
Comment (halx99) [inline extensions/DrawNodeEx/src/DrawNodeEx/DrawNodeEx.cpp]:
```suggestion
    ensureCapacityTriangle(vertex_count);
```

**Review by halx99 — CHANGES_REQUESTED** [review]
Please replace all `ensureCapacityGLTriangle` to `ensureCapacityTriangle`

**Comment by aismann** [comment]
> Please replace all `ensureCapacityGLTriangle` to `ensureCapacityTriangle`

I think the other also:
```
    ensureCapacityGLTriangle(512);
    ensureCapacityGLPoint(64);
    ensureCapacityGLLine(256);
```
    
   replace to:
```
    ensureCapacityTriangle(512);
    ensureCapacityPoint(64);
    ensureCapacityLine(256);
```

**Comment by paulocoutinhox** [comment]
Hi, nice. Can you explain to me what is this to i understand?

**Comment by aismann** [comment]
> Hi, nice. Can you explain to me what is this to i understand?

https://github.com/axmolengine/axmol/issues/1579
The goal is to replace `class DrawNode` with this extension (will be renamed to DrawNode and be no longer an extension).

**Comment by aismann** [comment]
@halx99 
Plz merge

File: extensions/DrawNodeEx/src/DrawNodeEx/DrawNodeEx.cpp [hunk 6]
```diff
@@ -48,13 +48,13 @@ NS_AX_EXT_BEGIN
 
 static inline Tex2F v2ToTex2F(const Vec2& v)
 {
```
Comment (halx99) [inline extensions/DrawNodeEx/src/DrawNodeEx/DrawNodeEx.cpp]: Since tex2f, vec2 is same in axmol, all v2ToTex2F can be removed

**Review by halx99 — CHANGES_REQUESTED** [review]

File: extensions/DrawNodeEx/src/DrawNodeEx/DrawNodeEx.cpp (hunk 6 above)
Comment (aismann) [reply extensions/DrawNodeEx/src/DrawNodeEx/DrawNodeEx.cpp]: done

**Comment by aismann** [comment]
@halx99 
If there nothing more, you can merge plz.

**Review by halx99 — APPROVED** [review]

**Pull request merged**